src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -0,0 +1,915 @@
+      styles.left = '0';
+      styles.right = '';
+      styles.width = '100%';
+    } else {
+      styles.width = `${boundingBoxRect.width}px`;
+      styles.left = boundingBoxRect.left ? `${boundingBoxRect.left}px` : '';
+      styles.right = boundingBoxRect.right ? `${boundingBoxRect.right}px` : '';
+    }
+
+    const maxHeight = this._overlayRef.getConfig().maxHeight;
+    if (maxHeight && this._hasFlexibleHeight) {
+      styles.maxHeight = formatCssUnit(maxHeight);
+    }
+
+    const maxWidth = this._overlayRef.getConfig().maxWidth;
+    if (maxWidth && this._hasFlexibleWidth) {
+      styles.maxWidth = formatCssUnit(maxWidth);
+    }
+
+    this._lastBoundingBoxHeight = boundingBoxRect.height;
+    this._lastBoundingBoxWidth = boundingBoxRect.width;
+
+    extendStyles(this._boundingBox!.style, styles);
+  }
+
+  /** Resets the styles for the bounding box so that a new positioning can be computed. */
+  private _resetBoundingBoxStyles() {
+    extendStyles(this._boundingBox!.style, {
+      top: '0',
+      left: '0',
+      right: '0',
+      bottom: '0',
+      height: '',
+      width: '',
+      alignItems: '',
+      justifyContent: '',
+    } as CSSStyleDeclaration);
+  }
+
+  /** Sets positioning styles to the overlay element. */
+  private _setOverlayElementStyles(originPoint: Point, position: ConnectedPosition): void {
+    // Reset styles from any previous positioning.
+    const styles = {
+      top: '',
+      left: '',
+      bottom: '',
+      right: '',
+    } as CSSStyleDeclaration;
+
+    // Align the overlay panel to the appropriate edge of the
+    // size-constraining container unless using a 'center' position.
+    if (this._hasFlexibleWidth && position.overlayX !== 'center' && !this._isPushed) {
+      if (this._isRtl()) {
+        styles[position.overlayX === 'end' ? 'left' : 'right'] = '0';
+      } else {
+        styles[position.overlayX === 'end' ? 'right' : 'left'] = '0';
+      }
+    }
+
+    if (this._hasFlexibleHeight && position.overlayY !== 'center' && !this._isPushed) {
+      styles[position.overlayY === 'bottom' ? 'bottom' : 'top'] = '0';
+    }
+
+    if (!this._hasFlexibleHeight || this._isPushed) {
+      extendStyles(styles, this._getExactOverlayY(position, originPoint));
+    }
+
+    if (!this._hasFlexibleWidth || this._isPushed) {
+      extendStyles(styles, this._getExactOverlayX(position, originPoint));
+    }
+
+    // Use a transform to apply the offsets. We do this because the `center` positions rely on
+    // being in the normal flex flow and setting a `top` / `left` at all will completely throw
+    // off the position. We also can't use margins, because they won't have an effect in some
+    // cases where the element doesn't have anything to "push off of". Finally, this works
+    // better both with flexible and non-flexible positioning.
+    let transformString = '';
+
+    if (position.offsetX) {
+      transformString += `translateX(${position.offsetX}px)`;
```

Comment:
prefix with `' '` to prevent potential future bugs (the `trim` will just remove it anyways). or use an array and `.join(' ')`